File: test-signin.py
import json
import requests

print('🧪 Testing frontend sign-in flow...\n')

# Step 1: Login via backend
print('1️⃣ Testing backend login endpoint...')
login_res = requests.post('http://localhost:8000/api/auth/login/', json={
    'username': 'testuser@example.com',
    'password': 'TestPassword123'
})

if login_res.status_code != 200:
    print('❌ Backend login failed:', login_res.json())
    exit(1)

login_data = login_res.json()
access = login_data['access']
print('✅ Backend login successful')
print(f'   Access token: {access[:40]}...')

# Step 2: Test protected endpoint
print('\n2️⃣ Testing /auth/me/ with access token...')
me_res = requests.get('http://localhost:8000/api/auth/me/', headers={
    'Authorization': f'Bearer {access}'
})

if me_res.status_code != 200:
    print('❌ Protected endpoint failed:', me_res.json())
    exit(1)

user = me_res.json()
print('✅ Protected endpoint access successful')
print(f'   User: {user["email"]} ({user["role"]})')
print(f'   Verified: {user["is_email_verified"]}')

# Step 3: Check frontend homepage
print('\n3️⃣ Testing frontend homepage...')
home_res = requests.get('http://localhost:3001/', allow_redirects=False)
print(f'   Status: {home_res.status_code}')

# Step 4: Check dashboard page
print('\n4️⃣ Testing /dashboard page...')
dash_res = requests.get('http://localhost:3001/dashboard', allow_redirects=False)
print(f'   Status: {dash_res.status_code}')

print('\n✨ Backend sign-in flow test complete!')
print('\n📝 Next manual steps:')
print('   1. Open http://localhost:3001 in your browser')
print('   2. Click "Sign In" button')
print('   3. Enter: testuser@example.com / TestPassword123')
print('   4. Verify the app signs in successfully')
print('   5. Check the Network tab in browser DevTools for API calls')

File: test-signin.js
#!/usr/bin/env node

const http = require('http');

/**
 * Test the frontend sign-in flow by simulating what the browser auth-context.tsx does
 */

function makeRequest(method, url, headers = {}, body = null) {
  return new Promise((resolve, reject) => {
    const options = new URL(url);
    const client = url.startsWith('https') ? require('https') : http;
    
    const req = client.request(
      {
        method,
        hostname: options.hostname,
        port: options.port,
        path: options.pathname + options.search,
        headers: { 'Content-Type': 'application/json', ...headers },
      },
      (res) => {
        let data = '';
        res.on('data', (chunk) => { data += chunk; });
        res.on('end', () => {
          try {
            resolve({ status: res.statusCode, data: JSON.parse(data), headers: res.headers });
          } catch {
            resolve({ status: res.statusCode, data, headers: res.headers });
          }
        });
      }
    );
    
    req.on('error', reject);
    if (body) req.write(JSON.stringify(body));
    req.end();
  });
}

async function testSignIn() {
  console.log('🧪 Testing frontend sign-in flow...\n');
  
  // Step 1: Login via backend
  console.log('1️⃣ Testing backend login endpoint...');
  const loginRes = await makeRequest('POST', 'http://localhost:8000/api/auth/login/', {}, {
    username: 'testuser@example.com',
    password: 'TestPassword123'
  });
  
  if (loginRes.status !== 200) {
    console.error('❌ Backend login failed:', loginRes);
    process.exit(1);
  }
  
  const { access, refresh } = loginRes.data;
  console.log('✅ Backend login successful');
  console.log(`   Access token: ${access.substring(0, 40)}...`);
  
  // Step 2: Test protected endpoint with token
  console.log('\n2️⃣ Testing /auth/me/ with access token...');
  const meRes = await makeRequest('GET', 'http://localhost:8000/api/auth/me/', {
    'Authorization': `Bearer ${access}`
  });
  
  if (meRes.status !== 200) {
    console.error('❌ Protected endpoint failed:', meRes);
    process.exit(1);
  }
  
  console.log('✅ Protected endpoint access successful');
  console.log(`   User: ${meRes.data.email} (${meRes.data.role})`);
  console.log(`   Verified: ${meRes.data.is_email_verified}`);
  
  // Step 3: Test dashboard access
  console.log('\n3️⃣ Testing frontend homepage...');
  const homeRes = await makeRequest('GET', 'http://localhost:3001/');
  if (homeRes.status === 200) {
    console.log('✅ Frontend homepage loaded');
  } else {
    console.log('⚠️  Homepage returned:', homeRes.status);
  }
  
  // Step 4: Test dashboard page
  console.log('\n4️⃣ Testing /dashboard page...');
  const dashboardRes = await makeRequest('GET', 'http://localhost:3001/dashboard');
  if (dashboardRes.status === 200) {
    console.log('✅ Dashboard page loaded');
  } else {
    console.log('⚠️  Dashboard returned:', dashboardRes.status);
  }
  
  console.log('\n✨ Sign-in flow test complete!\n');
  console.log('📝 Next steps:');
  console.log('   1. Open http://localhost:3001 in your browser');
  console.log('   2. Click "Sign In" button');
  console.log('   3. Enter: testuser@example.com / TestPassword123');
  console.log('   4. Verify the app navigates to dashboard or home');
  console.log('   5. Check browser console for any auth errors');
}

testSignIn().catch(err => {
  console.error('❌ Test failed:', err);
  process.exit(1);
});
